### PR TITLE
APPS-788 fix project-wide reuse

### PR DESCRIPTION
**Pull request by jdidion**
Also improves `CompilerTest` to clean use a separate folder and clean it up at the end of the suite.

### CODE REVIEW

**Review by r-i-v-a — CHANGES_REQUESTED** [review]
-- Some tests are failing related to reorg
-- If this change is only in dxCompiler code, would like to include it in the release

**Comment by jdidion** [comment]
tests are passing now

File: compiler/src/main/scala/dx/compiler/DxExecutableDirectory.scala [hunk 1]
```diff
@@ -75,13 +75,16 @@ case class DxExecutableDirectory(bundle: Bundle,
...
+      folder: Option[String] = None,
+      recurse: Boolean = false
+  ): Vector[(DxDataObject, DxObjectDescribe)] = {
     Vector("applet", "workflow")
       .flatMap { dxClass =>
         val constraints = DxFindDataObjectsConstraints(
             project = Some(project),
             folder = folder,
-            recurse = false,
+            recurse = recurse,
```
Comment (emiloslavsky) [inline compiler/src/main/scala/dx/compiler/DxExecutableDirectory.scala:87]: does recursion happen within the folder or in the entire project (which is what should happen)?

Comment (jdidion) [reply compiler/src/main/scala/dx/compiler/DxExecutableDirectory.scala:87]: Entire project, unless a folder is specified (which it is not when `-projectWideReuse` is set)

**Review by r-i-v-a — APPROVED** [review]

**Pull request merged**